src/main/utils/nugetUtils.ts

```diff
@@ -0,0 +1,90 @@
+                { base: workspace.uri.fsPath, pattern: '**/*.sln' },
+                ScanUtils.getScanExcludePattern(workspace)
+            );
+            wsSolutions.forEach(solution => solutions.add(solution));
+        }
+        return Promise.resolve(solutions);
+    }
+    
+    /**
+     * @param workspaceFolders - Base workspace folders
+     * @param componentsToScan - Set of nuget components to populate during the tree building. We'll use this set later on, while scanning the packages with Xray.
+     * @param scanCacheManager - Scan cache manager
+     * @param parent           - The base tree node
+     * @param quickScan        - True to allow using the scan cache
+     */
+    public static async createDependenciesTrees(
+        workspaceFolders: vscode.WorkspaceFolder[],
+        componentsToScan: Collections.Set<ComponentDetails>,
+        treesManager: TreesManager,
+        parent: DependenciesTreeNode,
+        quickScan: boolean
+    ): Promise<NugetTreeNode[]> {
+        let solutions: Collections.Set<vscode.Uri> = await NugetUtils.locateSolutions(workspaceFolders, treesManager.logManager);
+        if (solutions.isEmpty()) {
+            treesManager.logManager.logMessage('No *.sln files found in workspaces.', 'DEBUG');
+            return [];
+        }
+    
+        treesManager.logManager.logMessage('Solution files to scan: [' + solutions.toString() + ']', 'DEBUG');
+        let nugetTreeNodes: NugetTreeNode[] = [];
+        for (let solution of solutions.toArray()) {
+            let tree: any = await NugetUtils.getProjects(solution.fsPath, treesManager.logManager, quickScan);
+            if (!tree) {
+                continue;
+            }
+            for (let project of tree.projects) {
+                let dependenciesTreeNode: NugetTreeNode = new NugetTreeNode(path.dirname(solution.fsPath), componentsToScan, treesManager, parent);
+                dependenciesTreeNode.refreshDependencies(quickScan, project);
+                nugetTreeNodes.push(dependenciesTreeNode);
+            }
+        }
+        return nugetTreeNodes;
+
+    }
+
+    private static async getProjects(slnFilePath: string, logManager: LogManager, quickScan: boolean) : Promise<any> {
+        let nugetList: any;
+        try {
+            nugetList = JSON.parse(NugetDepsTree.buildTree(slnFilePath));
+        } catch (error) {
+            logManager.logError(error, !quickScan);
+            logManager.logMessage('Failed building tree for solution "' + slnFilePath + '", due to the above error. Skipping to next solution... ', 'INFO');
+            return null;
+        }
+
+        if (!nugetList.projects) {
+            logManager.logError(new Error('No projects found for solution "' + slnFilePath + '".'), !quickScan);
+            logManager.logMessage(
+                'Possible cause: The solution needs to be restored. Restore it by running "nuget restore ' + slnFilePath + '.',
```

Comment:
```suggestion
                'Possible cause: The solution needs to be restored. Restore it by running "nuget restore" from "' + slnFilePath + '".',

```

Reply:
slnFilePath is a direct absolute path to the solution so the proposed command is correct (nuget restore /path/to/example.sln)

Reply:
OK, but it looks like there is still a missing `"` in the end

Reply:
Right, thanks.